internal/controllers/machineset/machineset_delete_policy.go

```diff
@@ -35,6 +35,7 @@ type (
 
 const (
 	mustDelete    deletePriority = 100.0
+	shouldDelete  deletePriority = 75.0
```

Comment:
While this might be a fine change on its own, regardless of the cause, could we have a reproducible test that shows the original bug?

Reply:
Technically the test that I have added already shows the bug when ran against current main. It will fail because the `nodeHealthyConditionUnknownMachine` would be picked by `getMachinesToDeletePrioritized` instead of the expected `mustDeleteMachine`.

Reply:
@ctrox Would it be, or would it be random? (we're not using a stable sort)

Reply:
You are right, the test I added previously did not take the name into account. I now added two cases with different names and one of them fails against current main since it would pick the `unhealthyMachineA` instead of the `mustDeleteMachine`.

Reply:
Perfect, thx!